src/custom/state/lists/reducer.ts

```diff
@@ -0,0 +1,201 @@
+          const currentActiveListsSet = chainId && new Set(state.activeListUrls)
+          // activate if on default active
+          // TODO: should this default to MAINNET if no ChainID?
+          if (defaultActiveListsSet?.has(url) && !currentActiveListsSet?.has(url)) {
+            state.activeListUrls?.push(url)
+          }
+
+          state.byUrl[url] = {
+            ...state.byUrl[url],
+            loadingRequestId: null,
+            error: null,
+            current: tokenList,
+            pendingUpdate: null
+          }
+        }
+      })
+      .addCase(fetchTokenList.rejected, (state, { payload: { url, requestId, errorMessage } }) => {
+        if (state.byUrl[url]?.loadingRequestId !== requestId) {
+          // no-op since it's not the latest request
+          return
+        }
+
+        state.byUrl[url] = {
+          ...state.byUrl[url],
+          loadingRequestId: null,
+          error: errorMessage,
+          current: null,
+          pendingUpdate: null
+        }
+      })
+      .addCase(addList, (state, { payload: url }) => {
+        if (!state.byUrl[url]) {
+          state.byUrl[url] = NEW_LIST_STATE
+        }
+      })
+      .addCase(removeList, (state, { payload: url }) => {
+        if (state.byUrl[url]) {
+          delete state.byUrl[url]
+        }
+        // remove list from active urls if needed
+        if (state.activeListUrls && state.activeListUrls.includes(url)) {
+          state.activeListUrls = state.activeListUrls.filter(u => u !== url)
+        }
+      })
+      .addCase(enableList, (state, { payload: url }) => {
+        if (!state.byUrl[url]) {
+          state.byUrl[url] = NEW_LIST_STATE
+        }
+
+        if (state.activeListUrls && !state.activeListUrls.includes(url)) {
+          state.activeListUrls.push(url)
+        }
+
+        if (!state.activeListUrls) {
+          state.activeListUrls = [url]
+        }
+      })
+      .addCase(disableList, (state, { payload: url }) => {
+        if (state.activeListUrls && state.activeListUrls.includes(url)) {
+          state.activeListUrls = state.activeListUrls.filter(u => u !== url)
+        }
+      })
+      .addCase(acceptListUpdate, (state, { payload: url }) => {
+        if (!state.byUrl[url]?.pendingUpdate) {
+          throw new Error('accept list update called without pending update')
+        }
+        state.byUrl[url] = {
+          ...state.byUrl[url],
+          pendingUpdate: null,
+          current: state.byUrl[url].pendingUpdate
+        }
+      })
+      .addCase(
+        initialiseTokenLists,
+        (state, { payload: { byUrl, lastInitializedDefaultListOfLists, activeListUrls } }) => {
+          state.byUrl = byUrl
+          state.lastInitializedDefaultListOfLists = lastInitializedDefaultListOfLists
+          state.activeListUrls = activeListUrls
+        }
+      )
+  // .addCase(updateVersion, state => {
```

Comment:
why is this commented? Isn't this for handling list updates?

Reply:
it's replaced by the logic inside the `updater` where it can leverage the `chainId`, i left it commented as a Mod reference

Reply:
I think I don't understand anymore the updater. I left some comments. 

When you accept a list and the list changes, Uniswap notifies in the UI. Did you test that this functionality is still working? I was kind of thinking the removed code had to do with that. 

In which easy way we could test that? maybe modifying the local storage and removing one token?

Reply:
it doesn't do what you think, it's just fired once when the app reloads and before it renders as per:
![Screenshot from 2021-02-25 19-35-34](https://user-images.githubusercontent.com/21335563/109207264-d65fb000-77a0-11eb-8e59-9823ded0e74d.png)